### PR TITLE
KAFKA-4774. Inner classes which don't need a reference to the outer c…

### PR DESCRIPTION
**Pull request by cmccabe**
…lass should be static

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/StickyTaskAssignor.java [hunk 1]
```diff
@@ -253,7 +253,7 @@ Pair pair(final TaskId task1, final TaskId task2) {
             return new Pair(task2, task1);
         }
 
-        class Pair {
+        static class Pair {
```
Comment (hachikuji) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/StickyTaskAssignor.java]: Seems this can be private?

Comment (cmccabe) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/StickyTaskAssignor.java]: Yeah

File: streams/src/main/java/org/apache/kafka/streams/state/internals/NamedCache.java [hunk 2]
```diff
@@ -320,7 +320,7 @@ synchronized void close() {
     /**
      * A simple wrapper class to implement a doubly-linked list around MemoryLRUCacheBytesEntry
      */
-    class LRUNode {
+    static class LRUNode {
```
Comment (ijuma) [inline streams/src/main/java/org/apache/kafka/streams/state/internals/NamedCache.java:323]: Seems like this one can be private too?

Comment (hachikuji) [reply streams/src/main/java/org/apache/kafka/streams/state/internals/NamedCache.java:323]: I think it's used in `NamedCacheTest`

**Pull request closed without merging**